Spatial.gd

```diff
@@ -490,101 +495,88 @@ func init_category_tree():
 
 
 func waypoint_categories_to_godot_nodes():
-	for category in self.waypoint_data.get_category():
-		_waypoint_categories_to_godot_nodes(null, category, category.get_name(), false)
+	for waypoint_category in self.waypoint_data.get_category():
+		_waypoint_categories_to_godot_nodes(null, waypoint_category, self.categories, self.minimap, false)
 
 
-func _waypoint_categories_to_godot_nodes(item: TreeItem, category, full_category_name: String, collapsed: bool):
+func _waypoint_categories_to_godot_nodes(item: TreeItem, waypoint_category: Waypoint.Category, parent_category: Spatial, parent_category2d: Node2D, collapsed: bool):
+	var godot_category = category_scene.instance()
+	var godot_category2d = category2d_scene.instance()
+	parent_category.add_subcategory(godot_category)
+	parent_category2d.add_subcategory(godot_category2d)
+
 	var category_item: TreeItem = self.marker_packs.create_item(item)
-	if category.get_name() == "":
-		# If this is called, there is an error in the Waypoint data
-		category_item.set_text(0, "No name")
-		category_item.set_metadata(0, "")
+	var category_data = CategoryData.new()
+	category_data.waypoint_category = waypoint_category
+	category_data.category = godot_category
+	category_data.category2d = godot_category2d
+
+	category_item.set_metadata(0, category_data)
+	if waypoint_category.get_name() == "":
 		print("Category found with no name.")
-		return
-	category_item.set_text(0, category.get_display_name())
-	category_item.set_metadata(0, full_category_name)
+		category_item.set_text(0, "No Name")
+	else:
+		category_item.set_text(0, waypoint_category.get_name())
```

Comment:
lets set a variable before this if statement, and then if that variable is empty we update it, instead of branching logic for what values to pass into `set_text()`